misk-feature/src/test/kotlin/misk/feature/testing/FakeFeatureFlagsTest.kt

```diff
@@ -321,4 +322,41 @@ internal class FakeFeatureFlagsTest {
     TYRANNOSAURUS,
     TALARURUS
   }
+
+  @Test
+  fun `trackString works`() {
+    subject.override(FEATURE, "test")
+    assertThat(subject.getString(FEATURE, TOKEN)).isEqualTo("test")
+    var valueReceivedByListenerReceived: String? = null
+    fun listener(newValue: String) {
+      require(valueReceivedByListenerReceived == null)
+      valueReceivedByListenerReceived = newValue
+    }
+
+    subject.trackString(FEATURE, Executors.newSingleThreadExecutor(),  ::listener)
+
+    // override the featureflag, this should trigger trackString
+    subject.override(FEATURE, "newValue")
+    Thread.sleep(10)
```

Comment:
This can make this test a bit flaky. Could you use some `CountDownLatch` like this [example](https://github.com/cashapp/misk/blob/45dfef7622b6d14dbd6281722c1f9cbd3279decc/misk-aws/src/test/kotlin/misk/jobqueue/sqs/SqsJobQueueServiceTest.kt#L58C18-L58C32).

Reply:
this is great, thank you!